graylog2-server/src/main/java/org/graylog2/users/UserImpl.java

```diff
@@ -198,20 +216,24 @@ public void setHashedPassword(final String hashedPassword) {
     }
 
     @Override
-    public void setPassword(final String password, final String passwordSecret) {
+    public void setPassword(final String password) {
         if (password == null || "".equals(password)) {
             // If no password is given, we leave the hashed password empty and we fail during validation.
             setHashedPassword("");
         } else {
-            final String newPassword = new SimpleHash(HASH_ALGORITHM, password, passwordSecret).toString();
+            final String newPassword = passwordAlgorithmFactory.defaultPasswordAlgorithm().hash(password);
             setHashedPassword(newPassword);
         }
     }
 
     @Override
-    public boolean isUserPassword(final String password, final String passwordSecret) {
-        final String oldPasswordHash = new SimpleHash(HASH_ALGORITHM, password, passwordSecret).toString();
-        return getHashedPassword().equals(oldPasswordHash);
+    public boolean isUserPassword(final String password) {
+        final PasswordAlgorithm passwordAlgorithm = passwordAlgorithmFactory.forPassword(getHashedPassword());
+        if (passwordAlgorithm == null) {
```

Comment:
This could be collapsed to `return passwordAlgorithm != null && passwordAlgorithm.matches(getHashedPassword(), password);`